### DISCUSSION TITLE
Where does dbt work end?

### DISCUSSION THREAD

**Discussion by christineberger** [dbt-labs/devhub-writing#58] [Guides and playbooks]
This is a question asked often - and even though we have various places where we attempt to explain how we think of modeling in dbt, it still seems unclear. The aim of this article would be to run through an example looking at some code. 

High level:
There are two high-level goals that can be seen in some dbt projects:
Concept Building: Explain the unfiltered nature of raw -> final concepts
Question Answering: Explain the filtered nature of final concepts -> reports

When do you tend to see this scenario?
Many people have this question whether they choose to do the question answering in dbt or in the BI tool, but mainly this is seen in projects which choose to preaggregate data in dbt before moving into their BI tool for visualization. A good example is Tableau users, where doing joins and aggregating isn't always efficient when done from the tool, so the choice is made to do the work from dbt first. 

Diving in to details:
1. Showing an example project which has both areas and explaining where we would place transformations that we see happening
2. Explain how we use organization to cleanly denote where concept building and question answering are happening
3. Explain how we use naming conventions to make apparent what our goals are of each step in concept building
4. Explain how we use naming conventions to make apparent which models can answer specific questions

**Comment by christineberger** [comment]
Not sure if this exactly fits the "definitive guide" category, but it *is* about data modeling. In the end people can choose how to do things, and might do it in a way that helps in the specific way their team and workflow functions. There's no wrong answers - this is just a general guide and what we've seen work well!

**Reply by krevitt** [reply]
Love this topic!  @patkearns10 had already drafted a similar post (Tableau-focused also) [in this issue](https://github.com/dbt-labs/dbt-technical-blog-writing/issues/21), [his notes start here in notion](https://www.notion.so/dbtlabs/Model-Naming-Conventions-d4d7d6ef284545d9827836662f7ab01f#7a372e080082462cbe4817e060e8f61f).

We ended up publishing part 1 (on the importance of naming conventions), but there's a part 2 lurking there (tentative title was "beyond marts").  

Maybe the two of you should collab on that part 2 post?
